src/HTTP/UrlClient.cpp

```diff
@@ -76,7 +75,7 @@ class cUrlClientRequest:
 		{
 			return nullptr;
 		}
-		cX509CertPtr cert;
+		cX509CertPtr cert = std::make_shared<cX509Cert>();
```

Comment:
Why? C++ has default constructors...

Reply:
Indeed it does. `std::shared_ptr`'s default constructor doesn't allocate anything while `std::make_shared` does.

Reply:
Oh, men... Just thought, that empty parameters to `make_shared` would create an empty shard_ptr...